### PR TITLE
Tracer particles for Advection_AmrCore

### PR DESCRIPTION
**Pull request by etpalmer63**
## Summary
Adds tracer particles to the `Advection_AmrCore` test. 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Tests/Amr/Advection_AmrCore/Source/AmrCoreAdv.cpp [hunk 1]
```diff
@@ -161,16 +172,18 @@ AmrCoreAdv::InitData ()
         InitFromScratch(time);
         AverageDown();
 
+#ifdef AMREX_PARTICLES
+    init_particles();
```
Comment (ax3l) [inline Tests/Amr/Advection_AmrCore/Source/AmrCoreAdv.cpp]: More consistent spaces:
```suggestion
        init_particles();
```

File: Tests/Amr/Advection_AmrCore/Source/AmrCoreAdv.cpp [hunk 2]
```diff
@@ -161,16 +172,20 @@ AmrCoreAdv::InitData ()
         InitFromScratch(time);
         AverageDown();
 
+#ifdef AMREX_PARTICLES
+    if (do_tracers) {
+        init_particles();
+    }
```
Comment (ax3l) [inline Tests/Amr/Advection_AmrCore/Source/AmrCoreAdv.cpp]: Alignment with code above is a bit off:
```suggestion
        if (do_tracers) {
            init_particles();
        }
```

**Comment by etpalmer63** [comment]
Particle and grid data output look good with `ref_ratio = 2`. Next I'll look at what is needed for general `ref_ratio`.

**Comment by etpalmer63** [comment]
Rebased and moved to #2722

**Pull request closed without merging**